### PR TITLE
Add a plugin and test for imageio

### PR DESCRIPTION
**Pull request by blink1073**
From the test with optional dependencies (it installed and is passing tests):

```
skimage.io.tests.test_imageio.TestSave.test_imsave_roundtrip(array([[139, 182, 153, 138, 108, 164, 111, 227, 245,  97], ... ok
skimage.io.tests.test_imageio.TestSave.test_imsave_roundtrip(array([[[172,  68, 187], ... ok
skimage.io.tests.test_imageio.TestSave.test_imsave_roundtrip(array([[[102, 236,  25, 241], ... ok
skimage.io.tests.test_imageio.test_imageio_flatten ... ok
skimage.io.tests.test_imageio.test_imageio_palette ... ok
skimage.io.tests.test_imageio.test_imageio_truncated_jpg ... ok
```

cc @almarklein.

### CODE REVIEW

**Comment by almarklein** [comment]
Sweet!

**Comment by almarklein** [comment]
cc @Kuchi

**Comment by blink1073** [comment]
From the test with dependencies:

`skimage.io.tests.test_imageio.test_imageio_truncated_jpg ... ok`

File: skimage/io/tests/test_imageio.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import os.path
+import numpy as np
+from numpy.testing import *
+from numpy.testing.decorators import skipif
+
+from tempfile import NamedTemporaryFile
+
+from skimage import data_dir
+from skimage.io import imread, imsave, use_plugin, reset_plugins
+import skimage.io as sio
```
Comment (ahojnnes) [inline skimage/io/tests/test_imageio.py]: Is this needed? You only use it for imread which is already imported.

**Comment by ahojnnes** [comment]
LGTM except for one small comment.

File: skimage/io/tests/test_imageio.py (hunk 1 above)
Comment (blink1073) [inline skimage/io/tests/test_imageio.py]: Fixed.

**Comment by ahojnnes** [comment]
In we go.

**Comment by ahojnnes** [comment]
Nope, I lied... :-) Travis does not like you.

**Comment by blink1073** [comment]
Yeah something happened to our 2.6 build, see #1692.

**Comment by ahojnnes** [comment]
Wanted to merge, but this PR has conflicts...

**Comment by blink1073** [comment]
Rebased, waiting for Travis again...

**Comment by blink1073** [comment]
Fixed, ready to roll.

**Comment by ahojnnes** [comment]
Thanks Steven.

**Pull request merged**